pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/JsonAsyncHttpPinotClientTransportFactory.java

```diff
@@ -85,8 +85,9 @@ public JsonAsyncHttpPinotClientTransportFactory withConnectionProperties(Propert
       _headers = ConnectionUtils.getHeadersFromProperties(properties);
     }
 
-    if (_scheme == null) {
-      _scheme = properties.getProperty("scheme", CommonConstants.HTTP_PROTOCOL);
+    String scheme = properties.getProperty("scheme", CommonConstants.HTTP_PROTOCOL);
+    if (_scheme == null || !_scheme.contentEquals(scheme)) {
+      _scheme = scheme;
```

Comment:
isn't this just a fancy way to say:
```suggestion
    _scheme = properties.getProperty("scheme", CommonConstants.HTTP_PROTOCOL);
```